test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/function/BiPredicateTest.scala

```diff
@@ -0,0 +1,127 @@
+    assertFalse(evenAndInRange.test(21, 42)) // Left Range true, Left Not Even
+    assertFalse(evenAndInRange.test(42, 41)) // Left Range true, Right Not Even
+    assertFalse(evenAndInRange.test(5, 22))  // Right Range true, Left Not Even
+    assertFalse(evenAndInRange.test(5, 21))  // Right Range true, Right Not Even
+
+    // Short-circuit
+    assertFalse(largerThan10LessThan30.and(dontCallPredicate).test(5, 5))   // Left false, Right true
+    assertFalse(largerThan10LessThan30.and(dontCallPredicate).test(20, 40)) // Left true, Right false
+    assertFalse(largerThan10LessThan30.and(dontCallPredicate).test(5, 40))  // Both false
+
+    // Left false, Right true
+    assertThrows(classOf[ThrowingPredicateException],
+      throwingPredicate.and(dontCallPredicate).test(5, 5))
+
+    // Left true, Right false
+    assertThrows(classOf[ThrowingPredicateException],
+      throwingPredicate.and(dontCallPredicate).test(20, 40))
+
+    // Both false
+    assertThrows(classOf[ThrowingPredicateException],
+      throwingPredicate.and(dontCallPredicate).test(5, 40))
+
+  }
+
+  @Test def negate(): Unit = {
+    val notLeftLargerThan10AndRightLessThan30 = largerThan10LessThan30.negate()
+    assertTrue("Right false", notLeftLargerThan10AndRightLessThan30.test(20, 40))
+    assertTrue("Left false", notLeftLargerThan10AndRightLessThan30.test(5, 20))
+    assertTrue("Both false", notLeftLargerThan10AndRightLessThan30.test(5, 40))
+
+    assertFalse("Both true", notLeftLargerThan10AndRightLessThan30.test(40, 20))
+  }
+
+  @Test def or(): Unit = {
+    // Truth table
+    val evenOrLargerThan10 = largerThan10LessThan30.or(even)
+
+    assertFalse("Left false, Both Odd", evenOrLargerThan10.test(5, 21))
+    assertFalse("Right false, Both Odd", evenOrLargerThan10.test(41, 41))
+    assertFalse("Both false, Both Odd", evenOrLargerThan10.test(5, 41))
+    assertTrue("Both true, Both Odd", evenOrLargerThan10.test(41, 21))
+
+    assertFalse("Left false, Left Even", evenOrLargerThan10.test(6, 21))
+    assertFalse("Left false, Right Even", evenOrLargerThan10.test(5, 22))
+    assertFalse("Right false, Left Even", evenOrLargerThan10.test(42, 41))
+    assertFalse("Right false, Right Even", evenOrLargerThan10.test(41, 42))
+    assertTrue("Both false, Both Even", evenOrLargerThan10.test(6, 42))
+
+    assertTrue("Both true, Left even", evenOrLargerThan10.test(42, 21))
+    assertTrue("Both true, Right even", evenOrLargerThan10.test(41, 22))
+    assertTrue("Both true, Both even", evenOrLargerThan10.test(42, 22))
+
+    // Short-circuit
+    assertTrue(largerThan10LessThan30.or(dontCallPredicate).test(42, 22))
+
+    assertThrows(classOf[ThrowingPredicateException],
+      throwingPredicate.or(dontCallPredicate).test(42, 22))
+  }
+}
+
+object BiPredicateTest {
+  final class ThrowingPredicateException(x: Any)
+    extends Exception(s"throwing predicate called with $x")
+
+  def makeBiPredicate[T, U](f: T => Boolean, f2: U => Boolean): BiPredicate[T, U] = {
```

Comment:
You probably won't need this with SAM's :)

Reply:
I think I stole this code style/naming from `Predicate` and was trying to make `BiPredicate` consistent with the non-Bi test: https://github.com/scala-js/scala-js/blob/829c2a322903805641e79251875260ec1e61fae2/test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/function/PredicateTest.scala#L88